csharp/src/Google.Protobuf/UnsafeCollectionOperations.cs

```diff
@@ -0,0 +1,118 @@
+#region Copyright notice and license
+
+// Protocol Buffers - Google's data interchange format
+// Copyright 2008 Google Inc.  All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Security;
+using Google.Protobuf.Collections;
+
+namespace Google.Protobuf
+{
+    /// <summary>
```

Comment:
I think the info here should be merged with the comment on `AsSpan`.

The overall comment for the type could be similar to CollectionMarshal: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.collectionsmarshal?view=net-8.0

> An unsafe class that provides a set of methods to access the underlying data representations of collections.

The reason to simplify it is because in the further there could be methods that are specific to `MapField`, and then the current class comment wouldn't make sense.

Reply:
+1